MaxText/train.py

```diff
@@ -71,10 +71,12 @@ def validate_train_config(config):
 
 # https://arxiv.org/pdf/2204.02311.pdf Appendix B
 def calculate_training_tflops(num_model_parameters, config):
+  """ Calculate training TFLOP"""
   learnable_weight_tflops = 6 * num_model_parameters * config.max_target_length * config.per_device_batch_size \
                                    / 10**12
   attention_tflops = 12 * config.num_heads * config.num_decoder_layers * config.head_dim * config.max_target_length**2 \
                      * config.per_device_batch_size / 10**12
+  attention_tflops = attention_tflops / 2 # due to causality in attention
```

Comment:
nit: I think it is bad form to rename here. Call the first var "noncasual_attention_flops" and the second one "causal_attention_flops".